src/Workspaces/Core/MSBuild/MSBuild/Build/ProjectBuildManager.cs

```diff
@@ -147,23 +101,49 @@ internal class ProjectBuildManager
             }
         }
 
+        public Task<(MSB.Evaluation.Project project, DiagnosticLog log)> LoadProjectAsync(
+            string path, CancellationToken cancellationToken)
+        {
+            if (_started)
+            {
+                return LoadProjectAsync(path, _projectCollection, cancellationToken);
+            }
+            else
+            {
+                var projectCollection = new MSB.Evaluation.ProjectCollection(AllGlobalProperties());
+                try
+                {
+                    return LoadProjectAsync(path, projectCollection, cancellationToken);
+                }
+                finally
+                {
+                    // unload project so collection will release global strings
+                    projectCollection.UnloadAllProjects();
+                }
+            }
+        }
+
         public async Task<string> TryGetOutputFilePathAsync(
-            string path, IDictionary<string, string> globalProperties, CancellationToken cancellationToken)
+            string path, CancellationToken cancellationToken)
         {
-            // This tries to get the project output path and retrieving the $(TargetPath) property.
+            Debug.Assert(_started);
 
-            var (project, _) = await LoadProjectAsync(path, globalProperties, cancellationToken).ConfigureAwait(false);
+            // This tries to get the project output path and retrieving the evaluated $(TargetPath) property.
+
+            var (project, _) = await LoadProjectAsync(path, cancellationToken).ConfigureAwait(false);
             return project?.GetPropertyValue(PropertyNames.TargetPath);
         }
 
-        public void Start()
+        public void Start(IDictionary<string, string> globalProperties = null)
```

Comment:
I don't understand why this take in new properties instead of just using `AllGlobalProperties()`.